dsl/src/main/scala/org/http4s/dsl/impl/Path.scala

```diff
@@ -385,6 +385,22 @@ abstract class OptionalValidatingQueryParamDecoderMatcher[T: QueryParamDecoder](
     }
 }
 
+sealed abstract class BadQuery
+
+object BadQuery {
+  final case class MissingRequiredParam(param: String) extends BadQuery
+  final case class InvalidParam(param: String, errors: NonEmptyList[ParseFailure]) extends BadQuery
+
+  implicit class ops[A](self: ValidatedNel[BadQuery, A]) {
```

Comment:
> Does BadQuery being a type param get BadQuery's companion into implicit scope? 

I think so, yeah. Actually, we should be able to do something similar with `Message` to avoid needing an import for the `Kleisli` syntax.